pkg/southbound/clientManager.go

```diff
@@ -233,39 +205,39 @@ func Subscribe(target Target, request *gpb.SubscribeRequest, handler client.Noti
 	q.TLS = target.Destination.TLS
 	q.NotificationHandler = handler
 	c := client.New()
-	err = c.Subscribe(target.Ctxt, q, "")
```

Comment:
why do you need to pass the ctx and can't use the one saved in the target ? Same in all other methods.

Reply:
I followed Go rules for using context in a struct vs passing it as an argument to the functions. Here (https://golang.org/pkg/context/) recommends to pass it as an argument. If you search the web, you will find discussions around this topic and some of the potential problems that can be raised if define it in a struct. I read somewhere if we pass it using channels between processes then it is OK to define in it in the struct. Northbound interface also followed the same paradigm (i.e. passing it as an argument). Personally, I would prefer to have abstractions to minimize number of arguments that a user needs to provide to a function but I think it would be better to follow the rules.

Reply:
ok. is it fine if we read through a structure and always pass that into the method ? is that safe ?

Reply:
Here (https://medium.com/@cep21/how-to-correctly-use-context-context-in-go-1-7-8f2c0fafdf39) explains when that would be OK to use it inside struct. I still need to go thorough that article and some other documents to see how can we apply it to our case. It looks like we need to use channels for communication between functions to make sure that is safe.

Reply:
After reading a little bit about this topic, I think the current implementation that we pass context as an argument to the functions is the best option at this moment. If you have any ideas, please let me know to make it better.

Reply:
I agree with Adib. The docs are fairly explicit about passing the context directly. This also makes it clear to the maintainers that a function is part of a request handling chain.